### ISSUE TITLE
Import, display and filter by Label/Publisher

### ISSUE THREAD

**Issue by Baswazz** [navidrome/navidrome#389]
Would be nice if the label/publisher is present in the id3 tag it is visible in the web ui.
When you click on it it filters on all releases by label, like when you click on the artist.

**Comment by deluan** [comment]
That's a good idea, thanks! I'll be adding more tags to be imported soon (release type, musicbrainz ids, etc...) and will start showing these tags in the UI and add filter for some of them. As the columns of the song grid are not configurable right now, I may not include `label` as a column for now, but I may add it as a filter, and it will be present in the album/song details

**Comment by certuna** [comment]
I believe this is the `TPUB` frame in id3v2.x, although MusicBrainz might write its own custom tag?

**Comment by deluan** [comment]
@certuna Navidrome uses `ffmpeg` to read all metadata (tags), so this is what ffmpeg reports:

- For id3 tags:
```
Input #0, mp3, from './Nine Inch Nails/The Slip/01-01 999,999.mp3':
  Metadata:
    title           : 999,999
    artist          : Nine Inch Nails
    album           : The Slip
    publisher       : The Null Corporation
```
- For Vorbis comments:
```
Input #0, flac, from './Nine Inch Nails/Ghosts I–IV/01-01 1 Ghosts I.flac':
  Metadata:
    ALBUM           : Ghosts I–IV
    ARTIST          : Nine Inch Nails
    TITLE           : 1 Ghosts I
    LABEL           : The Null Corporation
    PUBLISHER       : The Null Corporation
```
(I left out irrelevant fields)

So that should be easy. Just don't know what to do with the Vorbis example above: Should I use the `Label` or the `Publisher` field? Thoughts?

**Comment by certuna** [comment]
I'd say read both label and publisher, make it a multi-value field?

**Comment by deluan** [comment]
Yeah, that's what I've been thinking: combine all values from both fields, removing duplicates.

**Comment by certuna** [comment]
Looking into this - as with all text fields, id3v2.4 and Vorbis allow the label field to have multiple values. This in principle means this can't just be stored in the `mediaFile` table as a simple `label` field, we need a `label` table.

**Comment by certuna** [comment]
Still relevant and valuable

**Comment by arnaudgst** [comment]
So how about display and filtering label / Publisher in 2023 ? i just installed navidrome, but manage to show them. Is it already active ? do i need to trig something ? many thanks for your help and good work

**Comment by paradajz** [comment]
Chiming in here. This would be very useful.

**Comment by andylundqvist** [comment]
I also wonder about this, @deluan responded positively above but I am not sure if we can already filter by or display the publisher tag somehow? This would be a fantastic addition to Navidrome if it was implemented.

**Comment by metalheim** [comment]
> I also wonder about this, @deluan responded positively above but I am not sure if we can already filter by or display the publisher tag somehow? This would be a fantastic addition to Navidrome if it was implemented.

Not yet, will probably have to wait for the New Scanner #2709

**Comment by andylundqvist** [comment]
This looks amazing! I had missed it completely. Thank you for the information.

**Comment by crispybegs** [comment]
was there any progress on this? i have a large dance music collection i want to dedicate a navidrome instance to, and that's all based around record labels as much as artists (if not more so, actually). I'd love the ability to filter by record label, as well as year in the UI, and have similar options in the sidebar.

**Comment by deluan** [comment]
> was there any progress on this? i have a large dance music collection i want to dedicate a navidrome instance to, and that's all based around record labels as much as artists (if not more so, actually). I'd love the ability to filter by record label, as well as year in the UI, and have similar options in the sidebar.

I'm trying to get this done for the general testing phase of #2709, probably end of this week.

**Comment by deluan** [comment]
Done!

![Image](https://github.com/user-attachments/assets/53efad0f-ccd1-4201-b660-da529d911216)

**Comment by crispybegs** [comment]
amazing, thanks!!!

this at least means i can organise playlists by record label as well. is there a way of showing tracks in playlists in a graphical grid format as in the main view like this:

<img width="612" alt="Image" src="https://github.com/user-attachments/assets/c3245d86-9507-4dc1-87cb-0d048047582b" />

instead of this?

<img width="936" alt="Image" src="https://github.com/user-attachments/assets/559d7592-5cb7-4b27-9448-f826727f465d" />

**Comment by arnaudgst** [comment]
> Done!
> 
> ![Image](https://github.com/user-attachments/assets/53efad0f-ccd1-4201-b660-da529d911216)

Oh My God !!! Many Thanks ! I was Waiting for this for months ! Now in my Opinion, Navidrome Rock !!!

**Comment by deluan** [comment]
> this at least means i can organise playlists by record label as well. is there a way of showing tracks in playlists in a graphical grid format as in the main view like this:

I plan to do this in the rewritten UI, the next big sub-project for Navidrome. But this feature won't be implemented in the current UI.

**Comment by awsms** [comment]
> I plan to do this in the rewritten UI, the next big sub-project for Navidrome.

Has the rewriting of the UI already begun?

**Comment by deluan** [comment]
> > I plan to do this in the rewritten UI, the next big sub-project for Navidrome.
> 
> Has the rewriting of the UI already begun?

Yes, but it will be a while before a public version is available.
